Detectors/ITSMFT/MFT/condition/include/MFTCondition/DCSConfigUtils.h

```diff
@@ -47,34 +48,36 @@ class DCSConfigUtils
   {
   }
 
-  auto getAddress(const std::string name, const std::string type)
+  int getAddress(std::string name, std::string type)
   {
-    return static_cast<int>((mNameDictType[type])[name]);
+    std::map<std::string, int> dict = mNameDict[type];
+    return static_cast<int>(dict[name]);
   }
 
-  const std::string getName(int add, const std::string type)
+  std::string getName(int add, std::string type)
   {
-    return static_cast<const std::string>((mAddressDictType[type])[add]);
+    std::map<int, std::string> dict = mAddressDict[type];
+    return static_cast<std::string>(dict[add]);
   }
 
-  std::map<int, const std::string>& getAddressMap(const std::string type)
+  std::map<int, std::string> getAddressMap(std::string type)
   {
-    return mAddressDictType[type];
+    return mAddressDict[type];
   }
 
-  std::map<int, const std::string>& getAddressMap(int type)
+  std::map<int, std::string> getAddressMap(int type)
   {
-    return mAddressDictType[getTypeName(type)];
+    return mAddressDict[getTypeName(type)];
   }
 
-  std::map<const std::string, int>& getNameMap(const std::string type)
+  std::map<std::string, int> getNameMap(std::string type)
   {
-    return mNameDictType[type];
+    return mNameDict[type];
   }
 
-  std::map<const std::string, int>& getNameMap(int type)
+  std::map<std::string, int> getNameMap(int type)
   {
-    return mNameDictType[getTypeName(type)];
+    return mNameDict[getTypeName(type)];
   }
```

Comment:
All these methods return copies of the maps, are you sure this is what you want and not references?

Reply:
This class is used to get information that can be understood by a human (just an interpreter). In the initiDictionaly, the dictionaries (maps) are created and then it is used to translate machine-like information, e.g. addresses, to human-kind information, e.g. variable name. So, this is what I want to do.

Reply:
I mean `std::map<std::string, int> getNameMap(int type)` creates/returns a copy of the `mNameDict[getTypeName(type)]` at each call instead of returning a reference on the existing map. Sorry, I don't understand what it has to do with the type of information. Anyway, this does not prevent the class from working, can be changed later.